### PR TITLE
Add script to format files and update contribution guidelines

### PR DESCRIPTION
**Pull request by achocron**
Things to consider before you submit the PR:

* [x] Are tests passing locally?
* [ ] ~Has `CHANGELOG.md` been updated?~
* [ ] ~Are the files formatted correctly?~
* [ ] ~Did you add unit tests?~
* [ ] ~Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?~

## Description

Added a script that uses clang-format and our custom format file to format all files that have changes against origin/develop. Updated readme. Make sure to run `brew install clang-format` before running the script.

### CODE REVIEW

File: clang-format-changed-files.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
```
Comment (guperrot) [inline clang-format-changed-files.sh]: Can't tell on github UI but please make that the `chmod u+x` for this file is part of the commit.

Comment (dhei) [reply clang-format-changed-files.sh]: Yes, we did that.

File: clang-format-changed-files.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+# This script runs clang_format on all .h, .m, and .mm files that have changes from develop branch.
+# Before running this, ensure that your local develop branch is up to date.
+git fetch
+i=0
+for modified_file in $(git diff origin/develop --name-only *.h *.m *.mm)
```
Comment (guperrot) [inline clang-format-changed-files.sh]: On forks origin/develop will likely be stale, while upstream/develop might be the right one to serve as a reference (and could be another name).

This makes this script highly unreliable. We might need to adapt logic to find the right remote to fetch and compare against.

Comment (guperrot) [reply clang-format-changed-files.sh]: Would do: UPSTREAM=`git remote -v | grep ":Microsoft" | awk '{print $1}' | uniq`
git fetch $UPSTREAM
git diff "$UPSTREAM/develop"

File: clang-format-changed-files.sh [hunk 3]
```diff
@@ -2,9 +2,10 @@
 
 # This script runs clang_format on all .h, .m, and .mm files that have changes from develop branch.
 # Before running this, ensure that your local develop branch is up to date.
-git fetch
+upstream=$(git remote -v | grep "Microsoft" | awk '{print $1}' | uniq)
```
Comment (guperrot) [inline clang-format-changed-files.sh]: nit: ":Microsoft/" to avoid matching users that would use Microsoft somewhere in their account or org.

**Pull request merged**